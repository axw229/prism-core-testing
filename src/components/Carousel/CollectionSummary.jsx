// @flow
import React, { PureComponent } from 'react'
import { fullColorNumber, getContrastYIQ } from '../../../src/shared/helpers/ColorUtils'
import type { CollectionsSummary } from '../../shared/types/Colors'
import cloneDeep from 'lodash/cloneDeep'
import './CollectionSummary.scss'
const baseClass = 'collection__summary'

type Props = {
  data: CollectionsSummary,
  getSummaryData: Function,
  isExpertColor: boolean,
  handlePrev: Function,
  handleNext: Function,
  itemsPerView: number,
  itemNumber: number
}

const KEY_CODE_TAB = 9
const KEY_CODE_ENTER = 13
const KEY_CODE_SPACE = 32

class CollectionSummary extends PureComponent<Props> {
  handleClick = () => {
    const { getSummaryData, data } = this.props
    getSummaryData(data)
  }
  handleKeyDown = (e) => {
    if (e.keyCode === KEY_CODE_ENTER || e.keyCode === KEY_CODE_SPACE) {
      e.preventDefault()
      this.handleClick()
    }

    if (!e.shiftKey && e.keyCode === KEY_CODE_TAB && ((this.props.itemNumber) % this.props.itemsPerView === 0)) {
      this.props.handleNext()
    } else if (e.shiftKey && e.keyCode === KEY_CODE_TAB && ((this.props.itemNumber) % this.props.itemsPerView === 1)) {
      this.props.handlePrev()
    }
  }
  render () {
    const { data, isExpertColor } = this.props
    let color = { hex: '', name: '', brandKey: '', colorNumber: '' }
    let img
    let collectionName
    let bottomColorList

    if (isExpertColor) {
      if (!data.colorDefs) return null
      color = cloneDeep(data.colorDefs[0])
      bottomColorList = cloneDeep(data.colorDefs.slice(1))
    } else {
      img = data.img
      collectionName = data.name
      bottomColorList = cloneDeep(data.collections.slice(0, 5))
    }

    return (
      <React.Fragment>
        <div className={`${baseClass}`}>
          <div className={`${baseClass}__wrapper`} role='button' tabIndex='0' onClick={this.handleClick} onKeyDown={this.handleKeyDown}>
            {isExpertColor &&
              <div className={`${baseClass}__top-section`} style={{ backgroundColor: color.hex, color: getContrastYIQ(color.hex) }}>
                <div className={`${baseClass}__content__wrapper`}>
                  <div className={`${baseClass}__content__wrapper__color-number`} >
                    {fullColorNumber(color.brandKey, color.colorNumber)}
                  </div>
                  <div className={`${baseClass}__content__wrapper__color-name`}>
                    {color.name}
                  </div>
                </div>
              </div>
            }
            {
              !isExpertColor &&
              <img className={`${baseClass}__top-section__image`} alt={`${collectionName}. ${data.description}`} src={img} />
            }
            <div className={`${baseClass}__bottom-list`}>
              {bottomColorList.map((color, key) => {
                return <div key={key} style={{ backgroundColor: color.hex }} className={`${baseClass}__bottom-item
              ${isExpertColor ? `${baseClass}__bottom-item--thick-border` : `${baseClass}__bottom-item--thin-border`}`} />
              })}
            </div>
          </div>
          <div>
            {
              !isExpertColor &&
              <div className={`${baseClass}__collection-name`}>
                {collectionName}
              </div>
            }
          </div>
        </div>
      </React.Fragment>
    )
  }
}

export default CollectionSummary
