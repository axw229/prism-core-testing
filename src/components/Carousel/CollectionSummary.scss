.collection__summary {
  padding: 2% 3%;
  flex: 50% 0 0;
  @include respond-above(sm){
    flex: 25% 0 0;
  }
  outline: none;
  &__wrapper {
    border: 3px solid #ffffff;
    cursor: pointer;
    outline:none;
    &:hover {
      border-color: #2cabe2;
    }
  }
  &__top-section {
    min-height: 100px;
    &__image {
      width: 100%;
      height: auto;
    }
  }
  &__content__wrapper {
    padding: 14px;
    &__color-name{
      font-weight: 700;
    }
  }
  &__bottom-item {
    flex: 1 1 0;
    border-top: 3px solid #ffffff;
    &--thin-border {
      &:not(:last-child) {
        border-right: 1px solid #ffffff;
      }
    }
    &--thick-border {
      &:not(:last-child) {
        border-right: 3px solid #ffffff;
      }
    }
  }
  &__bottom-list {
    display: flex;
    height: 30px;
  }
  &__collection-name {
    text-align: center;
    font-weight: 700;
    margin: 1.5em 0 0;
  }
}
