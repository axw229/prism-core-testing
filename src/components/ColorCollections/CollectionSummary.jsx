// @flow
import React, { PureComponent } from 'react'
import { fullColorNumber, getContrastYIQ } from '../../shared/helpers/ColorUtils'
import type { CollectionsSummary } from '../../shared/types/Colors.js.flow'
import cloneDeep from 'lodash/cloneDeep'
import './CollectionSummary.scss'
const baseClass = 'collection__summary'

type Props = {
  data: CollectionsSummary,
  getSummaryData: Function,
  isExpertColor: boolean
}

class CollectionSummary extends PureComponent<Props> {
  handleClick = () => {
    const { getSummaryData, data } = this.props
    getSummaryData(data)
  }
  render () {
    const { data, isExpertColor } = this.props
    let color = { hex: '', name: '', brandKey: '', colorNumber: '' }
    let img
    let collectionName
    let bottomColorList

    if (isExpertColor) {
      if (!data.colorDefs) return null
      color = cloneDeep(data.colorDefs[0])
      bottomColorList = cloneDeep(data.colorDefs.slice(1))
    } else {
      img = data.img
      collectionName = data.name
      bottomColorList = cloneDeep(data.collections.slice(0, 5))
    }

    return (
      <React.Fragment>
        <div className={`${baseClass}`}>
          <div className={`${baseClass}__wrapper`} role='button' tabIndex='-1' onClick={this.handleClick} onKeyDown={this.handleClick}>
            {isExpertColor &&
              <div className={`${baseClass}__top-section`} style={{ backgroundColor: color.hex, color: getContrastYIQ(color.hex) }}>
                <div className={`${baseClass}__content__wrapper`}>
                  <div className={`${baseClass}__content__wrapper__color-number`} >
                    {fullColorNumber(color.brandKey, color.colorNumber)}
                  </div>
                  <div className={`${baseClass}__content__wrapper__color-name`}>
                    {color.name}
                  </div>
                </div>
              </div>
            }
            {
              !isExpertColor &&
              <img className={`${baseClass}__top-section__image`} alt='' src={img} />
            }
            <div className={`${baseClass}__bottom-list`}>
              {bottomColorList.map((color, key) => {
                return <div key={key} style={{ backgroundColor: color.hex }} className={`${baseClass}__bottom-item
              ${isExpertColor ? `${baseClass}__bottom-item--thick-border` : `${baseClass}__bottom-item--thin-border`}`} />
              })}
            </div>
          </div>
          <div>
            {
              !isExpertColor &&
              <div className={`${baseClass}__collection-name`}>
                {collectionName}
              </div>
            }
          </div>
        </div>
      </React.Fragment>
    )
  }
}

export default CollectionSummary
