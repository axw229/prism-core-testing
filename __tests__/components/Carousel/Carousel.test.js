import React from 'react'
import { shallow } from 'enzyme'
import ListWithCarousel from 'src/components/Carousel/Carousel'
import CollectionSummary from 'src/components/ColorCollections/CollectionSummary'

const baseClass = 'prism-slick-carousel'
const contentWrapper = `${baseClass}__wrapper__content`
const createCarousel = (props = {}) => {
  const CollectionSummaryWrapper = ListWithCarousel(CollectionSummary)
  return shallow(<CollectionSummaryWrapper {...props} />)
}
const wrapper = createCarousel({ data: [] })

describe('snapshot match testing', () => {
  expect(wrapper).toMatchSnapshot()
})

describe('Carousel should rendering correctly', () => {
  it('should rendring wrapper component', () => {
    expect(wrapper.find(`${baseClass}__wrapper`)).toBeTruthy()
  })
  it('should rendring buttons correctly', () => {
    expect(wrapper.find(`${contentWrapper}__buttons`)).toBeTruthy()
  })
  it('should rendering child component correctly', () => {
    expect(wrapper.find('collection-list__component')).toBeTruthy()
  })
})

describe('Testing state for carousel component', () => {
  it('State should initialize correctly', () => {
    const initialState = wrapper.instance().state
    expect(initialState).toEqual({
      curr: 0,
      itemsPerView: 8,
      width: window.innerWidth
    })
  })
})

describe('Event tesing for behavoir of carousel component', () => {
  it('Testing for click buttons', () => {
    const nextClick = jest.fn()
    const prevClick = jest.fn()
    wrapper.instance().handleNext = nextClick
    wrapper.instance().handlePrev = prevClick
    wrapper.instance().forceUpdate()
    wrapper.find(`.${contentWrapper}__next-btn__wrapper >.${contentWrapper}__buttons`).simulate('click')
    wrapper.find(`.${contentWrapper}__prev-btn__wrapper >.${contentWrapper}__buttons`).simulate('click')
    expect(nextClick).toHaveBeenCalled()
    expect(prevClick).toHaveBeenCalled()
  })
})
