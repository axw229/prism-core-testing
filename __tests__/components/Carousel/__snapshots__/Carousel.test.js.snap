// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="prism-slick-carousel__wrapper"
>
  <div
    className="prism-slick-carousel__wrapper__content"
  >
    <div
      className="prism-slick-carousel__wrapper__content__prev-btn__wrapper"
    >
      <button
        className="prism-slick-carousel__wrapper__content__buttons prism-slick-carousel__wrapper__content__buttons--visible"
        onClick={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "fa",
              "chevron-left",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </button>
    </div>
    <div
      className="prism-slick-carousel__wrapper__content__list__wrapper "
    >
      <div
        className="collection-list__container  "
        style={
          Object {
            "transform": "translateX(-0%)",
          }
        }
      />
    </div>
    <div
      className="prism-slick-carousel__wrapper__content__next-btn__wrapper"
    >
      <button
        className="prism-slick-carousel__wrapper__content__buttons prism-slick-carousel__wrapper__content__buttons--visible"
        onClick={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "fa",
              "chevron-right",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </button>
    </div>
  </div>
  <div
    className="prism-slick-carousel__wrapper__indicators"
  />
</div>
`;
